### PR TITLE
[KNUDEV-6] implemented QueryDsl and had wroten tests for it

### LINKED ISSUE #6: Configure QueryDSL

### PR DESCRIPTION
**Pull request by DenysLnk**

### CODE REVIEW

File: task-manager/pom.xml [hunk 1]
```diff
@@ -17,7 +17,33 @@
...
+            <artifactId>querydsl-jpa</artifactId>
+            <version>5.1.0</version>
+            <classifier>jakarta</classifier>
+        </dependency>
+        <dependency>
+            <groupId>com.querydsl</groupId>
+            <artifactId>querydsl-apt</artifactId>
+            <version>5.1.0</version>
+            <classifier>jakarta</classifier>
+        </dependency>
```
Comment (cumbainer) [inline task-manager/pom.xml]: You already have `querydsl-jpa` and `querydsl-apt` in common module, so you can remove it from here

File: task-manager/pom.xml [hunk 2]
```diff
@@ -17,7 +17,33 @@
...
+            <groupId>org.easymock</groupId>
+            <artifactId>easymock</artifactId>
+            <version>5.4.0</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (cumbainer) [inline task-manager/pom.xml:34]: i guess here we can put dependency `spring-boot-starter-test` in parent pom, which have all this jupiter and stuff
so you can remove this dependency from here

File: knudev-common/src/main/java/ua/knu/knudev/knudevcommon/config/QEntityManagerUtil.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    private EntityManager entityManager;
+
+    private static JPAQueryFactory queryFactory;
+
+    @PostConstruct
+    private void init() {
+        queryFactory = new JPAQueryFactory(JPQLTemplates.DEFAULT, entityManager);
+    }
+
+//    todo not delete because with it we can make static calls to queryFactory
```
Comment (cumbainer) [inline knudev-common/src/main/java/ua/knu/knudev/knudevcommon/config/QEntityManagerUtil.java]: Remove this comment

File: task-manager/src/test/java/ua/knu/knudev/query/dsl/QueryDslTest.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
+package ua.knu.knudev.query.dsl;
```
Comment (cumbainer) [inline task-manager/src/test/java/ua/knu/knudev/query/dsl/QueryDslTest.java]: In Telegram we disscussed problems with this tests, will fix in future

File: team-manager/pom.xml [hunk 5]
```diff
@@ -9,8 +9,36 @@
...
     </parent>
-
     <artifactId>team-manager</artifactId>
 
+    <dependencies>
+        <dependency>
+            <groupId>ua.knu</groupId>
+            <artifactId>knudev-common</artifactId>
+        </dependency>
+        <dependency>
```
Comment (cumbainer) [inline team-manager/pom.xml]: As already said, you already have trhis in common moduke

File: team-manager/src/test/java/ua/knu/knudev/query/dsl/QueryDslTest.java [hunk 6]
```diff
@@ -0,0 +1,137 @@
+package ua.knu.knudev.query.dsl;
```
Comment (cumbainer) [inline team-manager/src/test/java/ua/knu/knudev/query/dsl/QueryDslTest.java]: why do you need this, when you already have same tests in task manager>

**Review by cumbainer — CHANGES_REQUESTED** [review]

File: task-manager/pom.xml (hunk 1 above)
Comment (DenysLnk) [reply task-manager/pom.xml]: ok

File: task-manager/pom.xml (hunk 2 above)
Comment (DenysLnk) [reply task-manager/pom.xml:34]: ok

File: knudev-common/src/main/java/ua/knu/knudev/knudevcommon/config/QEntityManagerUtil.java (hunk 3 above)
Comment (DenysLnk) [reply knudev-common/src/main/java/ua/knu/knudev/knudevcommon/config/QEntityManagerUtil.java]: ok

File: task-manager/src/test/java/ua/knu/knudev/query/dsl/QueryDslTest.java (hunk 4 above)
Comment (DenysLnk) [reply task-manager/src/test/java/ua/knu/knudev/query/dsl/QueryDslTest.java]: ok

File: team-manager/pom.xml (hunk 5 above)
Comment (DenysLnk) [reply team-manager/pom.xml]: ok

File: team-manager/src/test/java/ua/knu/knudev/query/dsl/QueryDslTest.java (hunk 6 above)
Comment (DenysLnk) [reply team-manager/src/test/java/ua/knu/knudev/query/dsl/QueryDslTest.java]: ok

**Pull request merged**